spec.bs

```diff
@@ -1068,34 +1260,79 @@ Note: Two {{PAHistogramContribution}}s for the same report can be merged if and
 
 <div algorithm>
 To <dfn>query the budget</dfn> given a [=list=] of {{PAHistogramContribution}}s
-|contributions|, an [=origin=] <var ignore>origin</var>, a [=context type=] <var
-ignore>api</var> and a [=moment=] <var ignore>currentTime</var> and a
-[=boolean=] |consumeIfPermitted|, perform the following steps. They return a
-[=list=] of [=booleans=] with the same [=list/size=] as |contributions|.
+|contributions|, a [=list=] of [=named budget ledgers=]
+|remainingBudgetLedgers|, an [=origin=] |origin|, a [=context type=] |api|, a
+[=moment=] |currentTime|, and a [=boolean=] |consumeIfPermitted|, perform the
+following steps. They return a [=list=] of [=booleans=] with the same
+[=list/size=] as |contributions|.
+
+1. If |remainingBudgetLedgers| is [=list/empty=]:
+    1. [=list/For each=] |budgetingScope| of [=all budgeting scopes=]:
+        1. Let |remainingBudgetLedger| be a new [=named budget ledger=].
+        1. [=map/Set=] |remainingBudgetLedger|[null] to |budgetingScope|'s
+            [=budgeting scope/max budget=].
+        1. [=list/Append=] |remainingBudgetLedger| to |remainingBudgetLedgers|.
+1. Let |numScopes| be the [=list/size=] of [=all budgeting scopes=].
+1. [=Assert=] that |numScopes| is equal to the [=list/size=] of
```

Comment:
Wondering if the ledgers should be a map or something if we go the route of actually enumerating the two 'scopes' (I.e. to avoid the parallel lists).

If we wanted to simplify this logic, we could also consider just computing allocations based off the 2^16 10-min limit and scaling it up 16x for the daily limit. We could also consider having the implementation-defined algorithm below just return a "budget left" for each name, taking into account both scopes. But it's up to you how much detail you want to put in this PR.